x-pack/plugins/ingest_pipelines/public/application/services/navigation.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+const BASE_PATH = '/';
+
+const EDIT_PATH = 'edit';
+
+const CREATE_PATH = 'create';
+
+const _getEditPath = (name: string, encode = true): string => {
+  return `${BASE_PATH}${EDIT_PATH}/${encode ? encodeURIComponent(name) : name}`;
+};
+
+const _getCreatePath = (): string => {
+  return `${BASE_PATH}${CREATE_PATH}`;
+};
+
+const _getClonePath = (name: string, encode = true): string => {
+  return `${BASE_PATH}${CREATE_PATH}/${encode ? encodeURIComponent(name) : name}`;
+};
+const _getListPath = (name?: string): string => {
+  return `${BASE_PATH}${name ? `?pipeline=${encodeURIComponent(name)}` : ''}`;
+};
+
+export const ROUTES_CONFIG = {
```

Comment:
Nit: I'm not sure the `CONFIG` suffix adds more information, since personally I'd infer `ROUTES` to be a constant, and from the name it would contain the various routes of the app.